File: java/Toasts/app/src/main/java/com/github/shellhub/toasts/MainActivity.java
package com.github.shellhub.toasts;

import android.os.Bundle;
import android.view.View;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    /**
     * Show Toasts when button clicked
     *
     * @param view deliver view
     */
    public void showToast(View view) {
        switch (view.getId()) {
            case R.id.btn_short_toasts:
                Toast.makeText(this, getResources().getText(R.string.short_toasts), Toast.LENGTH_SHORT).show();
                break;
            case R.id.btn_long_toasts:
                Toast.makeText(this, getResources().getText(R.string.long_toasts), Toast.LENGTH_SHORT).show();
                break;
            case R.id.btn_custom_toasts:
                View toastCustomView = getLayoutInflater().inflate(R.layout.custom_toasts, null);
                Toast customToast = new Toast(this);
                customToast.setView(toastCustomView);
                customToast.show();
                break;
            default:
                //TODO
        }
    }
}

File: kotlin/Toasts/app/src/main/java/com/github/shellhub/toast/MainActivity.kt
package com.github.shellhub.toast

import android.os.Bundle
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    /**
     * executed when button clicked
     */
    fun showToast(view: View) {
        when (view.id) {
            R.id.btn_short_toasts -> Toast.makeText(
                this,
                R.string.short_toasts,
                Toast.LENGTH_SHORT
            ).show()
            R.id.btn_long_toasts -> Toast.makeText(
                this,
                R.string.long_toasts,
                Toast.LENGTH_SHORT
            ).show()
            R.id.btn_custom_toasts -> {
                val inflater = layoutInflater
                val layout = inflater.inflate(
                    R.layout.custom_toasts,
                    findViewById(R.id.custom_toast_container)
                )
                with(Toast(applicationContext)) {
                    duration = Toast.LENGTH_LONG
                    this.view = layout
                    show()
                }
            }
            else -> {
                //TODO
            }

        }
    }
}
